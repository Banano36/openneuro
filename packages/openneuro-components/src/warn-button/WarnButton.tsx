import React, { useRef, useEffect } from 'react'
import './warn-button.scss'
import { Tooltip } from '../tooltip/Tooltip'
import { Button } from '../button/Button'

export interface WarnButtonProps {
  message?: string
  icon?: string
  disabled?: boolean
  tooltip?: string
  onConfirmedClick?: () => void
  displayOptions: boolean
  setDisplayOptions(cb: (currentState: boolean) => boolean): void
}
export const WarnButton = ({
  message,
  icon,
  disabled,
  tooltip,
  onConfirmedClick,
  displayOptions,
  setDisplayOptions,
}: WarnButtonProps) => {
  const viewAction = (
    <div className="warn-btn-group " role="group">
      <div className="slide-in">
        <Button
          className="btn-warn-component cancel"
          iconSize="12px"
          icon="fa fa-times"
          color="#fff"
          backgroundColor="#c00342"
          onClick={() => setDisplayOptions(currentState => !currentState)}
          size="xsmall"
        />

        <Button
          className="btn-warn-component success"
          iconSize="12px"
          icon="fa fa-check"
          color="#fff"
          backgroundColor="#00b489"
          onClick={() => {
            onConfirmedClick()
            setDisplayOptions(currentState => !currentState)
          }}
          size="xsmall"
        />
      </div>
    </div>
  )

  const hideAction = (
    <span className={disabled ? ' disabled warn-btn-click' : 'warn-btn-click'}>
      <Button
        className="btn-warn-component"
        iconSize="12px"
        icon={'fa ' + icon}
        label={message}
        onClick={() => setDisplayOptions(currentState => !currentState)}
        disabled={disabled}
      />
    </span>
  )

  const button = displayOptions ? viewAction : hideAction

  return (
    <div className="warn-btn">
      {tooltip ? (
        <Tooltip flow="up" tooltip={tooltip}>
          {button}
        </Tooltip>
      ) : (
        button
      )}
    </div>
  )
}
