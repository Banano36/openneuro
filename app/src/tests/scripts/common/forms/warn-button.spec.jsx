import WarnButton from '../../../../scripts/common/forms/warn-button';

describe('component WarnButton', () => {
    it('renders successfully', () => {
        const wrapper = shallow(
            <WarnButton message="A Button!"/>
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('renders with warnings disable', () => {
        const wrapper = shallow(
            <WarnButton message="A Button!" warn={false}/>
        );
        expect(wrapper).toMatchSnapshot();
    })
});
