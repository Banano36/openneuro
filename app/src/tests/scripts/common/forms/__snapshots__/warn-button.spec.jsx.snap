// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component WarnButton renders successfully 1`] = `
<span
  className="fade-in"
>
  <button
    className="btn-warn-component warning"
    onClick={[Function]}
  >
    <i
      className="fa fa-trash-o"
    />
      
    A Button!
  </button>
</span>
`;

exports[`component WarnButton renders with warnings disable 1`] = `
<span
  className="fade-in"
>
  <button
    className="btn-warn-component warning"
    onClick={[Function]}
  >
    <i
      className="fa fa-trash-o"
    />
      
    A Button!
  </button>
</span>
`;
